discovery_engine/lib/src/domain/feed_manager.dart

```diff
@@ -38,25 +40,46 @@ class FeedManager {
   /// Handle the given feed client event.
   ///
   /// Fails if [event] does not have a handler implemented.
-  Future<void> handleFeedClientEvent(FeedClientEvent event) async {
-    await event.maybeWhen(
-      feedRequested: () => restoreFeed(),
-      nextFeedBatchRequested: () => nextFeedBatch(),
-      feedDocumentsClosed: (ids) => _docMgr.deactivateDocuments(ids),
-      orElse: throw UnimplementedError('handler not implemented for $event'),
-    );
-  }
+  Future<EngineEvent> handleFeedClientEvent(FeedClientEvent event) =>
+      event.maybeWhen(
+        feedRequested: () => restoreFeed(),
+        nextFeedBatchRequested: () => nextFeedBatch(),
+        feedDocumentsClosed: (ids) => _docMgr
+            .deactivateDocuments(ids)
+            .then((_) => const EngineEvent.clientEventSucceeded()),
```

Comment:
also not for this PR, just to open up a discussion, but I feel like it's a bit confusing that we have this indirection and calling one manager from the other. This looks like a good reason to have dedicated usecases for every event (by "usecase" I mean something like a manager that handles only 1 event). On the other hand, that would be more boilerplate'y to initialize. Wdyt?

Reply:
yes, the dependency on the document manager here also makes testing a little more awkward.
@acrrd is it necessary to have it this way, or could the `deactivateDocuments` method be part of the feed manager, say?